MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+    [self.managedObjectContext hydrateStoreWithJSONAtPath:path attributeMappings:attributes forEntityName:@"DamageZone"];
+}
+
+- (void)loadMonsterDropData
+{
+    NSArray *dropList = [self loadArrayFromJsonFileNamed:@"monsterDrops"];
+    for (NSDictionary *drop in dropList) {
+        // Only add the drop if it is not already there
+        if (![self checkCountForEntityWithEntityName:@"MonsterDrop" withIDNumber:drop[@"_dropID"]]) {
+            NSEntityDescription *newDropDescription = [NSEntityDescription entityForName:@"MonsterDrop" inManagedObjectContext:self.managedObjectContext];
+            MonsterDrop *newDrop = [[MonsterDrop alloc] initWithEntity:newDropDescription insertIntoManagedObjectContext:self.managedObjectContext];
+            [newDrop setDropBasicValuesWithID:drop[@"_dropID"] method:drop[@"method"] quantity:drop[@"quantity"]];
+            
+            // If the Item for this drop already exists in the persistent store ...
+            Item *item = (Item *)[self getUniqueEntityWithEntityName:@"Item" withNameAttribute:drop[@"item_name"]];
+            if (item) {
+                // We need to hook up relationships appropriately.
+                newDrop.item = item;
+                [item.monsterSourceSet addObject:newDrop];
+            } else {
+                // Otherwise, we should create the item.
+                NSEntityDescription *newItemDescription = [NSEntityDescription entityForName:@"Item" inManagedObjectContext:self.managedObjectContext];
+                Item *newItem = [[Item alloc] initWithEntity:newItemDescription insertIntoManagedObjectContext:self.managedObjectContext];
+                [newItem.monsterSourceSet addObject:newDrop];
+                newItem.name = drop[@"item_name"];
+                newDrop.item = newItem;
+            }
+            
+            // If this is a monster drop ...
+            if (![drop[@"monster_name"] isEqualToString:@""]) {
+                // We need to add this drop to the monster's list, if it is not there already.
+                
+                // First we should ensure that the monster already exists in the persistent store
+                Monster *monster = (Monster *)[self getUniqueEntityWithEntityName:@"Monster" withIDNumber:drop[@"monster_id"]];
+                if (!monster) {
+                    NSLog(@"Monster not found! Returning early.");
+                    @throw @"Monster nto found!";
+                    return;
+                } else {
+                    [monster.dropSet addObject:newDrop];
+                    newDrop.monsterSource = monster;
+                    newDrop.rank = drop[@"rank"];
+                    newDrop.percent = drop[@"percent"];
+                }
+            }
+        }
+    }
+    
+    // We should now try to save the manged object context
+    if (![self didSuccessfullySaveContext]) {
```

Comment:
You might also not want to attempt the save if the load operation returned no JSON objects.